testament/testament.nim

```diff
@@ -794,14 +796,16 @@ proc main() =
     p.next
     processPattern(r, pattern, p.cmdLineRest.string, simulate)
   of "r", "run":
+    # "/pathto/tests/stdlib/nre/captures.nim" -> "stdlib" + "tests/stdlib/nre/captures.nim"
     var subPath = p.key.string
-    if subPath.isAbsolute: subPath = subPath.relativePath(getCurrentDir())
+    let nimRoot = currentSourcePath / "../.."
+      # makes sure points to this regardless of cwd or which nim is used to compile this.
+    doAssert existsDir(nimRoot/testsDir) # sanity check
```

Comment:
This is not a "sanity check", it's input validation. Assertions are for bugs.

Reply:
there are no user inputs here, only internal program logic: currentSourcePath, nimRoot, testsDir.

Reply:
@timotheecour 
This made `testament` completely broken for basic stuff on devel.

```console
$ choosenim update devel
   Updating devel
    Updated to #devel

$ mkdir tests

$ echo "doAssert 1 == 1" > tests/texample.nim

$ testament r tests/texample.nim 

Error: unhandled exception: testament.nim(798, 13) `dirExists(nimRoot / testsDir)` 
/home/runner/work/nightlies/nightlies/nim-1.5.1/tests/ doesn't exist! [AssertionDefect]

$
```

`currentSourcePath` is a compile time value.

Works perfect on CI, broken everywhere else. :sob: :disappointed_relieved:

Reply:
see https://github.com/timotheecour/Nim/issues/519; it works if you compile testament yourself instead of using pre-installed testament; it's tricky to solve as mentioned in that issue (pre-existing behavior also was buggy)

I suggest solution 3 in https://github.com/timotheecour/Nim/issues/519 which is the simplest

Reply:
=> https://github.com/nim-lang/Nim/pull/16767